woocommerce/pinterest-for-woocommerce
# Handle Connecting with a Personal account.

Closes #168 
Closes #169

### Changes proposed in this pull request

This PR handles the case where the user connects with a personal account. 

After the user has authenticated with a personal account, there are 2 flows:
1. The user has no Linked business account. In that case they are asked to Create a business account:
![image](https://user-images.githubusercontent.com/4016167/135123482-fb574e1f-e3a9-4e88-be4e-7b61a837ec46.png)

Clicking the "Create Business Account" button will sent the user to Pinterest to create a linked business account. 
That flow in Pinterest doesn't return the user back to the plugin, so to go around that we are refreshing the Authenticated user's Business accounts on window.focus. 

So after the user navigates back to the plugin's tab, the accounts list should be populated.

2. The user already has Linked business accounts. In that case the accounts list will be presented and the flow from this point on is the same as if a Linked business account was created in flow `#1`.
![image](https://user-images.githubusercontent.com/4016167/135124376-cf4dfea5-1c2d-4e89-b4a0-fc110702644e.png)

Clicking connect will sent the user to the required middleware URL that will switch to the given business account and then navigate the user once again to the Auth screen, 
![image](https://user-images.githubusercontent.com/4016167/135125192-10007b32-097f-4f69-8e12-86066836037c.png)

so that Auth will be completed with the selected business account:
![image](https://user-images.githubusercontent.com/4016167/135125297-472f0320-4825-41de-92fd-ffb4a5e4e1bf.png)

#### Screenshots


### Detailed test instructions

1. Start with a clean plugin installation and a new Pinterest account. 
2. Start the onboarding process and Connect to Pinterest.
3. Verify that the plugin asks you to create a Linked Business account. 
4. Click the button to create one, complete the process in Pinterest and return to plugin. 
5. Verify that you are presented with the list of business accounts. (Should be 1 account at this point).
6. Click "connect", to connect to the selected account. 
7. Verify that on the Authentication screen the Business account is selected. 
8. After returning to the onboarding guide, click the 1st step's label on the top of the screen to navigate to the "Connection" step and verify that you are now connected to a business account.
